TimingFunction for all `AnimationType` Fix #564

phimage opened the pull request:
If a timing function is defined for "spring" animation type, do not use damping and velocity 

see explanation in #564


demo app updated to be able to select the timing function all animation types

phimage committed:
If a timing function is defined for "spring" animation type, do not use damping and velocity Fix #564

JakeLin reviewed (approved):
👍👍👍Thanks for fixing that, they can't work together 😞

tbaranes reviewed (approved):
Awesome!

phimage committed:
Do not use not valid timing function for spring animation
Document about timing function and spring animation
Update CHANGELOG
Fix swiftlint warning about space
